src/transform/src/analysis.rs

```diff
@@ -1857,3 +1845,20 @@ mod cardinality {
         }
     }
 }
+
+mod common_lattice {
+    use crate::analysis::Lattice;
+
+    pub struct BoolLattice;
+
+    impl Lattice<bool> for crate::analysis::common_lattice::BoolLattice {
+        fn top(&self) -> bool {
+            true
+        }
+        fn meet_assign(&self, into: &mut bool, item: bool) -> bool {
+            let changed = *into && !item;
+            *into = *into && item;
+            changed
+        }
+    }
+}
```

Comment:
I know this wasn't documented beforehand, but it would be good to add some docs.